src/MivaAccess/Services/Orders/MivaOrdersService.cs

```
@@ -9,86 +10,91 @@
 		{
 		}
 
-		public async Task< IEnumerable< MivaOrder > > GetModifiedOrdersFromQueueAsync( string queueName, CancellationToken token, Mark mark = null )
+		public async Task<IEnumerable<MivaOrder>> GetModifiedOrdersFromQueueAsync(string queueName, Mark mark, CancellationToken cancellationToken)
 		{
-			if ( mark == null )
-				mark = Mark.CreateNew();
-
-			if ( token.IsCancellationRequested )
-			{
-				var exceptionDetails = CreateMethodCallInfo( base.Config.ApiBaseUrl, mark, additionalInfo: this.AdditionalLogInfo() );
-				var mivaException = new MivaException( string.Format( "{0}. Get modified orders request was cancelled", exceptionDetails ) );
-				MivaLogger.LogTraceException( mivaException );
-			}
+			LogIfCancellationTokenRequested(mark, cancellationToken);
 
-			var orders = new List< MivaOrder >();
+			var orders = new List<MivaOrder>();
 			var pageIndex = 0;
 
-			while( true )
+			while (true)
 			{
-				var ordersFromPage = await CollectOrdersFromPage( queueName, pageIndex, base.Config.OrdersPageSize, token, mark ).ConfigureAwait( false );
+				var ordersFromPage = await CollectOrdersFromPageAsync(queueName, pageIndex, base.Config.OrdersPageSize, mark, cancellationToken).ConfigureAwait(false);
 
-				if ( !ordersFromPage.Any() )
+				if (!ordersFromPage.Any())
 				{
 					break;
 				}
 
-				orders.AddRange( ordersFromPage );
+				orders.AddRange(ordersFromPage);
 				++pageIndex;
 			}
 
 			return orders;
 		}
 
-		private async Task< IEnumerable< MivaOrder > > CollectOrdersFromPage( string queueName, int pageIndex, int pageSize, CancellationToken token, Mark mark )
+		public async Task AcknowledgeOrdersAsync(IEnumerable<long> ordersIds, Mark mark, CancellationToken cancellationToken)
 		{
-			var request = new GetOrdersFromQueueRequest( base.Config.Credentials, queueName );
-			request.SetPage( pageIndex, pageSize );
+			var chunks = ordersIds.SplitInBatches(AcknowledgeOrdersBatchLimit);
+			foreach (var chunk in chunks)

```

@maxim-saltanov-skuvault Can we rename to something more descriptive? Perhaps, `foreach (var orderIdsPage in orderIdsPages)`. Otherwise, `AcknowledgeOrdersRequest(base.Config.Credentials, chunk);` is not immediately obvious at a glance

---

@maxim-saltanov-skuvault The original comment here is resolved (renaming for clarity).

About the `token.IsCancellationRequested` PR comment you mention, I did delete it since as you mentioned you kept the logic but just moved it. And about throwing on IsCancellationRequested (as we do for some integrations) instead of just logging an exception with the context and then letting the caller throw a generic "Task Cancelled" (no mark), it seems that as long as we log the mark on IsCancellationRequested, we should be able to find the detailed exception. So either way might be Ok. But if @chris-skuvault has a preference one way vs the other, we can follow that.